fiona/_env.pyx

```diff
@@ -410,8 +413,8 @@ cdef class GDALEnv(ConfigEnv):
                         path = PROJDataFinder().search()
 
                         if path:
-                            os.environ['PROJ_LIB'] = path
-                            log.debug("PROJ data not found in environment, set to %r.", path)
+                            log.debug("PROJ data not found in environment, setting to %r.", path)
+                            set_proj_data_search_path(path)
```

Comment:
Note: for GDAL versions < 3.0, this function sets PROJ_LIB as before. For GDAL >= 3.0, it calls a new OSRSetPROJSearchPaths function that overrides PROJ_LIB and is more safe in other ways. PROJ_LIB continues to be respected by PROJ 6, but there are issues on Windows with modifying the environment within a program.